atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/priority/NextTableToSweepProviderTest.java

```diff
@@ -96,6 +96,54 @@ public void calculatorReturnsManyTablesWithHighestPriority_thenProviderReturnsOn
         Assert.assertThat(tableToSweep.get(), anyOf(is(table("table2")), is(table("table3")), is(table("table4"))));
     }
 
+    @Test
+    public void calculatorReturnsNonMaximumForPriorityTable_thenProviderStillSelectsIt() {
+        givenPriority(table("table1"), 1.0);
+        givenPriority(table("table2"), 10_000.0);
+
+        whenGettingNextTableToSweepWithOverrides(ImmutableSweepPriorityOverrideConfig.builder()
```

Comment:
Small style note: To keep with the BDD-style tests here I think this would have been better as:

```
givenPriority(table("table1"), 1.0);
givenPriority(table("table2"), 10_000.0);
givenPriorityOverride(table("table1"));

whenGettingNextTableToSweep();
...
```

This makes the test easier to read (the state is described in the 'given' phase) and if we ever change how overrides are specified we only have to change the `givenPriorityOverride` and `givenBlacklisted` methods.

Not essential to change though.

Reply:
OK, agreed that the above works better. First time working with this style of testing - must say the tests are very readable!